applications/nrf_desktop/doc/usb_state.rst

```diff
@@ -102,23 +109,56 @@ For example, the file contents can look as follows:
+USB HID configuration in USB legacy stack
+-----------------------------------------
+
+For the USB legacy stack, the :ref:`CONFIG_DESKTOP_USB_STACK_LEGACY <config_desktop_app_options>` selects the required dependencies: :kconfig:option:`CONFIG_USB_DEVICE_STACK` and :kconfig:option:`CONFIG_USB_DEVICE_HID` Kconfig options.
+The number of HID-class USB instances in the USB legacy stack is specified by :kconfig:option:`CONFIG_USB_HID_DEVICE_COUNT`.
+The default value of this option is updated to match requirements for either an nRF Desktop peripheral or a dongle.
+
+Low latency device configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The module sets the default value of :kconfig:option:`CONFIG_USB_HID_POLL_INTERVAL_MS` to ``1``.
+For low latency of HID reports, the device requests a polling rate of 1 ms.
+
+Boot protocol configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The :ref:`CONFIG_DESKTOP_HID_BOOT_INTERFACE_MOUSE <config_desktop_app_options>` and :ref:`CONFIG_DESKTOP_HID_BOOT_INTERFACE_KEYBOARD <config_desktop_app_options>` Kconfig options are used to control support of HID boot reports in the nRF Desktop application.
+The module aligns the USB HID boot protocol configuration with the application configuration.
+The module sets the default value of the :kconfig:option:`CONFIG_USB_HID_BOOT_PROTOCOL` Kconfig option and calls the :c:func:`usb_hid_set_proto_code` function during initialization to set the USB HID boot interface protocol code.
+Common HID boot protocol code is used for all of the HID-class USB instances.
+
+USB HID configuration in USB next stack
+---------------------------------------
+
+For the USB next stack, the :ref:`CONFIG_DESKTOP_USB_STACK_NEXT <config_desktop_app_options>` selects the :kconfig:option:`CONFIG_USB_DEVICE_STACK_NEXT` Kconfig option.
+Every USB HID-class instance is configured through a separate DTS node compatible with ``zephyr,hid-device``.
+The DTS node configures, among others, the used HID boot protocol, the size of the longest HID input report, and the HID polling rate.
+Make sure to update the DTS configuration to match requirements of your application.
+nRF Desktop application defines the USB HID-class instances used by the USB next stack for all of the supported boards and file suffixes.
+See the existing configurations for reference.
+
+Defining USB HID-class instances in the DTS, by default, enables the :kconfig:option:`CONFIG_USBD_HID_SUPPORT` Kconfig option.
+
 USB wakeup configuration
 ========================
 
 The nRF Desktop device can work as a source of wakeup events for the host device if connected through the USB.
 To use the feature, enable the :ref:`CONFIG_DESKTOP_USB_REMOTE_WAKEUP <config_desktop_app_options>` option.
-This option selects the :kconfig:option:`CONFIG_USB_DEVICE_REMOTE_WAKEUP` to enable required dependencies in the USB stack.
+This option selects the :kconfig:option:`CONFIG_USB_DEVICE_REMOTE_WAKEUP` Kconfig option to enable required dependencies if USB legacy stack is used.
```

Comment:
Shouldn't we mention what happens in next-stack here?

Reply:
No additional configuration is applied there

Reply:
But just reading this doc I have no idea is wakeup events working for next-stack. I would clearly state it here.

Reply:
Ok, added one more sentence. Hope it helps

Reply:
I meant to add sentence that states that wakeup events are working on next-stack:
If the USB next stack is used, no additional Kconfig dependencies need to be selected. -> 
If the USB next stack is used,  wakeup events are enabled by default and do not need any option to turn them on.

Reply:
Discussed offline. Rephrased a bit.